### PR TITLE
Move to pancake

### PR DESCRIPTION
**Pull request by jakozaur**
Migrates to pancake SQL query generation.

Before merge:
- [x] do more testing with Kibana

After merge:
- [ ] more test cleanup
- [ ] remove `level` from `TranslateSqlResponseToJson(rows []QueryResultRow, level int) JsonMap`
- [ ] several other cleanups

### CODE REVIEW

**Review by nablaone — CHANGES_REQUESTED** [review]
One change. We should return an error when parsing aggregation has failed.

File: quesma/queryparser/query_parser.go [hunk 1]
```diff
@@ -54,46 +54,22 @@ func (cw *ClickhouseQueryTranslator) ParseQuery(body types.JSON) (*model.Executi
...
-				pancakeApplied = true
-			} else {
-				logger.WarnWithCtx(cw.Ctx).Msgf("Error parsing pancake queries: %v. Falling back to the standard implementation.", err)
-			}
+		if pancakeQueries, err := cw.PancakeParseAggregationJson(body, addCount); err == nil {
+			queries = append(queries, pancakeQueries...)
+			countQuery = nil
+		} else {
+			// TODO: Change to error if failed, but also ignore if missing
+			logger.WarnWithCtx(cw.Ctx).Msgf("Error parsing pancake queries: %v. Falling back to the standard implementation.", err)
```
[the remark this answers is not in the document]
Comment (jakozaur) [reply quesma/queryparser/query_parser.go:67]: Ok, let me see what I can do there.

File: quesma/testdata/dashboard-1/aggregation_requests.go [hunk 2]
```diff
@@ -375,89 +357,75 @@ var AggregationTests = []testdata.AggregationTestCase{
...
-					model.NewQueryResultCol("key", int64(1713957360000/30000)),
-					model.NewQueryResultCol("doc_count", 78),
-				}},
-			},
-		},
+		/*
+			ExpectedResults: [][]model.QueryResultRow{
+				{{Cols: []model.QueryResultCol{model.NewQueryResultCol("hits", uint64(4800))}}},
+				{
+					{Cols: []model.QueryResultCol{
```
Comment (trzysiek) [inline quesma/testdata/dashboard-1/aggregation_requests.go:364]: why remove everywhere, but leave here as comment?

Comment (jakozaur) [reply quesma/testdata/dashboard-1/aggregation_requests.go:364]: This test does not have pancake version, so comment might be useful to build pancake equivalent.

**Review by trzysiek — APPROVED** [review]

File: quesma/quesma/search_test.go [hunk 3]
```diff
@@ -31,11 +33,23 @@ const defaultAsyncSearchTimeout = 1000
 
 const tableName = model.SingleTableNamePlaceHolder
 
+var ConfigWithPancake = config.QuesmaConfiguration{IndexConfig: map[string]config.IndexConfiguration{
```
[the remark this answers is not in the document]
Comment (jakozaur) [reply quesma/quesma/search_test.go]: Renamed to `DefaultPancake` and remove `pancake` optimizer.

File: quesma/queryparser/query_parser.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakozaur) [reply quesma/queryparser/query_parser.go]: Ok, addressed easier part. We no longer print warn when aggregations are missing and printing error on error.

Though addressing broader error issue is task for another big PR.

**Pull request merged**